### ISSUE TITLE
some build warnings

### ISSUE THREAD

**Issue by raveit65** [zhuyaliang/group-service#2]
During packaging a RPM i see that build warnings
```
Configuring config.h using configuration
Compiler for C supports link arguments -Wl,--version-script,/home/rave/rpmbuild/BUILD/group-service-1.0.0/src/libgroupservice/symbol.map: YES
WARNING: Project targetting '>=0.46.0' but tried to use feature introduced in '0.50.0': install arg in configure_file
Configuring org.group.admin.service using configuration
Configuring group-admin-daemon.service using configuration
Build targets in project: 14
WARNING: Project specifies a minimum meson_version '>=0.46.0' but uses features which were added in newer versions:
 * 0.50.0: {'install arg in configure_file'}
Option buildtype is: plain [default: debugoptimized]
```
Can this be ignored?

**Comment by raveit65** [comment]
During compiling i found only one warning
```
../src/util.c: In function 'setup_loginuid':
../src/util.c:233:5: warning: ignoring return value of 'write', declared with attribute warn_unused_result [-Wunused-result]
  233 |     write (fd, id, strlen (id));
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by yetist** [comment]
> Can this be ignored?

if meson >= 0.50.0, this can be ignored.
if meson >= 0.46.0 and meson <= 0.50.0, an error will occur.

meson <= 0.50.0 does not support this line:
https://github.com/zhuyaliang/group-service/blob/73c0fa98dec3a5cc7b48a0be28de50984727e965/data/meson.build#L25

**Comment by zhuyaliang** [comment]
@yetist 
Use version 1.1.0, which removes compilation warnings

**Comment by raveit65** [comment]
Thanks, using group-service-1.1.0 and user-admin-1.4.1 fixes all build issues.
I think for fedora 28 i need to lower `if meson >= 0.50.0`, meson-0.50 i scurrently only for f30 and f29 available.
But this i can do in spec file of rpm.

You are using a version for include dir, is this really needed?
from my spec file:
```
%files devel
%dir %{_includedir}/group-service-1.1/
%{_includedir}/group-service-1.1/libgroupservice/
%{_libdir}/libgroup-service.so
%{_libdir}/pkgconfig/group-service.pc
```
I think versions are only used for major version bumps, e.g. gtk-2/-3, by other packages.
```
/usr/include/gtk-2.0
/usr/include/gtk-3.0
```
If this is needed can you use full version, 1.1.0 ?
Than i can use the version rpm-macro in spec file to avoid editing the spec file for every new release.
```
%dir %{_includedir}/group-service-%{version}/
%{_includedir}/group-service-%{version}/libgroupservice/
%{_libdir}/libgroup-service.so
%{_libdir}/pkgconfig/group-service.pc
```

**Comment by zhuyaliang** [comment]
@raveit65 You're right. I don't use versions as included directories right now. ```group-service-1.0``` will always be used as the top-level directory name in the future

**Comment by raveit65** [comment]
I started a review for fedora https://bugzilla.redhat.com/show_bug.cgi?id=1690482

**Comment by raveit65** [comment]
@zhuyaliang 
Thank, for latest changes. Include dir is now.
```
%{_includedir}/group-service-1.0/
```
But it looks that shared libraries has also a soname bump.
```
%{_libdir}/libgroup-service.so.1
%{_libdir}/libgroup-service.so.1.1.0
```
better like this ?
```
%{_libdir}/libgroup-service.so.1
%{_libdir}/libgroup-service.so.1.0.0
```
See discussion in review request.
https://bugzilla.redhat.com/show_bug.cgi?id=1690482#c4

**Comment by eclipseo** [comment]
It's a minor soname bump, we only care about major soname bump.

**Comment by zhuyaliang** [comment]
@raveit65 @eclipseo Since the next version 1.1.1, the dynamic link library will have a fixed name of 1.0.0.

**Comment by raveit65** [comment]
Cool, thanks for fixing that.

**Comment by raveit65** [comment]
> if meson >= 0.50.0, this can be ignored.
> if meson >= 0.46.0 and meson <= 0.50.0, an error will occur.

> meson <= 0.50.0 does not support this line:
> group-service/data/meson.build
> Line 25 in 73c0fa9
> install: true,

Fedora 28 ships only meson-0.47.2-1.fc28 and build fails for this reason.
https://koji.fedoraproject.org/koji/taskinfo?taskID=33759581
@zhuyaliang 
Is there be a workaround for this?

**Comment by raveit65** [comment]
Or can i simply revert this change for building for meson-0.47.2
```
-  meson_version : '>=0.46.0',
+  meson_version : '>=0.50.0',
```

**Comment by raveit65** [comment]
Ok, that fixes the build for f28.
https://src.fedoraproject.org/cgit/rpms/group-service.git/commit/?h=f28
https://koji.fedoraproject.org/koji/taskinfo?taskID=33760257

**Comment by zhuyaliang** [comment]
@raveit65 Thank you very much for your help。So let's keep the source file unchanged（ ```meson_version``` ）
